lib/internal/Magento/Framework/Locale/TranslatedLists.php

```diff
@@ -150,10 +157,20 @@ public function getOptionCurrencies()
         $currencies = (new CurrencyBundle())->get($this->localeResolver->getLocale())['Currencies'] ?: [];
         $options = [];
         $allowed = $this->_config->getAllowedCurrencies();
+        $selectedCurrencies = explode(
+            ',', $this->scopeConfig->getValue(
+                'system/currency/installed',
+                \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+            )
+        );
         foreach ($currencies as $code => $data) {
             if (!in_array($code, $allowed)) {
                 continue;
             }
+            if (!in_array($code, $selectedCurrencies)) {
```

Comment:
I think condition could be moved to separate protected function, in order to encapsulate it